google/cloud/bigquery/_helpers.py

```diff
@@ -41,31 +42,53 @@
     re.VERBOSE,
 )
 
-_MIN_BQ_STORAGE_VERSION = pkg_resources.parse_version("2.0.0")
+_MIN_BQ_STORAGE_VERSION = packaging.version.Version("2.0.0")
+_BQ_STORAGE_OPTIONAL_READ_SESSION_VERSION = packaging.version.Version("2.6.0")
 
 
-def _verify_bq_storage_version():
-    """Verify that a recent enough version of BigQuery Storage extra is installed.
+class BQStorageVersions:
+    def __init__(self):
+        self._installed_version = None
 
-    The function assumes that google-cloud-bigquery-storage extra is installed, and
-    should thus be used in places where this assumption holds.
+    @property
+    def installed_version(
+        self,
+    ) -> Union[packaging.version.LegacyVersion, packaging.version.Version]:
+        if self._installed_version is None:
+            from google.cloud import bigquery_storage
 
-    Because `pip` can install an outdated version of this extra despite the constraints
-    in setup.py, the the calling code can use this helper to verify the version
-    compatibility at runtime.
-    """
-    from google.cloud import bigquery_storage
+            self._installed_version = packaging.version.parse(
+                getattr(bigquery_storage, "__version__", "legacy")
+            )
 
-    installed_version = pkg_resources.parse_version(
-        getattr(bigquery_storage, "__version__", "legacy")
-    )
+        return self._installed_version
 
-    if installed_version < _MIN_BQ_STORAGE_VERSION:
-        msg = (
-            "Dependency google-cloud-bigquery-storage is outdated, please upgrade "
-            f"it to version >= 2.0.0 (version found: {installed_version})."
-        )
-        raise LegacyBigQueryStorageError(msg)
+    @property
+    def is_read_session_optional(self) -> bool:
+        return self.installed_version >= _BQ_STORAGE_OPTIONAL_READ_SESSION_VERSION
+
+    def verify_version(self):
+        """Verify that a recent enough version of BigQuery Storage extra is installed.
+
+        The function assumes that google-cloud-bigquery-storage extra is installed, and
+        should thus be used in places where this assumption holds.
+
+        Because `pip` can install an outdated version of this extra despite the constraints
+        in setup.py, the the calling code can use this helper to verify the version
```

Comment:
Oops, I now noticed you copied my typo, let's remove it.

```suggestion
        in `setup.py`, the calling code can use this helper to verify the version
```